rfd/0057-automatic-aws-server-discovery.md

```diff
@@ -0,0 +1,91 @@
+---
+authors: Alex McGrath (alex.mcgrath@goteleport.com
+state: draft
+---
+
+# RFD 57 - Automatic discovery and enrollment of AWS servers
+
+## What
+
+Proposes a way by which an auth service might automatically discover and
+register AWS EC2 instances.
+
+## Why
+
+Currently when adding a new AWS server, its required that Teleport be installed
+after the server has been provisioned which may be a slow process for
+organizations with large numbers of servers as it needs to be installed and then
+added to the teleport cluster
+
+With the changes described in this document, Teleport will be able to resolve
+the issues with adding AWS servers to Teleport clusters automatically.
+
+
+## Discovery
+
+At least one Teleport auth agent will need to be manually set up so it can can
+be used to discover aws nodes.
+
+Discovery can use a matcher similar to the `db_service/aws` matcher:
+
+```yaml
+auth_service:
+  enabled: "yes"
+  aws:
+  - types: ["ec2"]
+    regions: ["us-west-1"]
+    tags:
+      "teleport": "yes"
+```
+
+The agent will use EC2's `DescribeInstances` API in order to list
+instance[1]. This will require the teleport auth agent to include
+`ec2:DescribeInstances` as part of it's IAM permissions
+
+As with database AWS discovery, new EC2 nodes will discovered periodically and
+any new nodes will be added and existing instances updated/deleted.
+
+Node hostname will be used in order to avoid attempting to reinstall Teleport on
```

Comment:
Node hostnames could potentially collide, should something else be used to identify nodes other than the hostname?